### PR TITLE
feat: add "Reaching Consensus: Developing Fault Tolerant SMTs in Golang" post

### PR DESCRIPTION
**Pull request by zivkovicmilos**
## Description

This PR adds the "Reaching Consensus: Developing Fault Tolerant SMTs in Golang" blog post by @zivkovicmilos and @petar-dambovaliev, along with the accompanying graphics.

- libtm migrated to the monorepo in: https://github.com/gnolang/gno/pull/2424

### CODE REVIEW

File: posts/2024-06-13_reaching_consensus/README.md [hunk 1]
```diff
@@ -0,0 +1,475 @@
+---
+publication_date: 2024-06-13T00:00:00Z
```
Comment (leohhhn) [inline posts/2024-06-13_reaching_consensus/README.md:2]: Can you change this to today's date? 🙏

File: posts/2024-06-13_reaching_consensus/README.md [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+However, we want to tackle this question of Tendermint’s inner-workings by weaving in the theory from the paper titled *The latest gossip on BFT consensus*, and applying it in practice.
+
+On a surface-level, the Tendermint consensus engine has 3 states:
+- **Propose** - the *proposer* for the current view (height, round) sends out a *proposal* to all other validator nodes. Non-proposers wait for the proposal and verify it
+- **Prevote** - the validator nodes wait for 2F+1 `PREVOTE` messages, potentially locking on a proposal
+- **Precommit** - the validator nodes wait for 2F+1 `PRECOMMIT` messages, potentially saving the block to the block store, and starting a new consensus round (with a view `(h+1, 0)`)
+
+These states are not as straightforward as they’re made out to be — there is an added complication of *state timers* which guarantee protocol liveness, and other asynchronous situations like consensus round jumps within the same height. We will discuss these mechanisms later on, as they are critical to making sure Tendermint is ticking correctly.
+
+### I. Propose
```
Comment (leohhhn) [inline posts/2024-06-13_reaching_consensus/README.md:66]: Roman numerals should not have a period after them. I suggest just switching it out to Arabic numerals and keep the period.

File: posts/2024-06-13_reaching_consensus/README.md [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+}
+```
+
+The pattern is clear:
+- subscribe to messages of a specific type that are expected to appear in the current state
+- verify those messages as they come in, and check if conditions are met for a state transition
+- monitor the outer context, and the asynchronous timer context for stopping
+
+Keeping this in mind, we can develop the main run loop:
+```go
```
Comment (leohhhn) [inline posts/2024-06-13_reaching_consensus/README.md:348]: Missing newline
```suggestion

```go
```

File: posts/2024-06-13_reaching_consensus/README.md [hunk 4]
```diff
@@ -0,0 +1,475 @@
...
+
+[![valset](https://gnolang.github.io/blog/2024-06-13_reaching_consensus/src/thumbs/valset.png)](https://gnolang.github.io/blog/2024-06-13_reaching_consensus/src/valset.png)
+
+We mentioned earlier that consensus is a process in which nodes agree upon a specific value. We denote this actual protocol in which they agree upon a value as the *consensus engine*. 
+
+The consensus engine represents the internal, most critical process of the consensus algorithm - the state machine. It gives answers to questions like - who proposes the consensus value (in leader-based systems), how is validation done, but most importantly — what are the steps taken by validators to reach consensus?
+
+[![machine](https://gnolang.github.io/blog/2024-06-13_reaching_consensus/src/thumbs/machine.png)](https://gnolang.github.io/blog/2024-06-13_reaching_consensus/src/machine.png)
+
+### Terminology
```
Comment (leohhhn) [inline posts/2024-06-13_reaching_consensus/README.md:40]:
```suggestion
### Terminology

```

**Review by leohhhn — APPROVED** [review]
Looks good, left a few minor comments, thanks 🙏

**Comment by mvertes** [comment]
Sorry @zivkovicmilos, I applied suggestions from @leohhhn in your place. I've mistaken the post! It should not matter too much anyway 🫢

**Pull request merged**